connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -681,6 +693,70 @@ class SparkConnectPlanner(session: SparkSession) {
     }
   }
 
+  private def transformWindowExpression(window: proto.Expression.Window) = {
```

Comment:
While the code is nicely structured, it is quite wide, I'm wondering if we can restructure it a bit or add more doc.